model/src/main/java/hu/blackbelt/judo/meta/jsl/validation/JslDslValidator.xtend

```diff
@@ -886,19 +886,34 @@ class JslDslValidator extends AbstractJslDslValidator {
         }
     }
 
-    @Check
-    def checkForDuplicateNameForDeclaration(Declaration declaration) {
-    	if (!(declaration.eContainer instanceof ModelDeclaration)) return;
-    	
-        if (declaration instanceof FunctionDeclaration) return;
-        if (declaration instanceof LambdaDeclaration) return;
+//    @Check
+//    def checkForDuplicateNameForDeclaration(Declaration declaration) {
+//    	if (!(declaration.eContainer instanceof ModelDeclaration)) return;
+//    	
+//        if (declaration instanceof FunctionDeclaration) return;
+//        if (declaration instanceof LambdaDeclaration) return;
+//
+//        if ((declaration.eContainer as ModelDeclaration).getDeclarationNames(declaration).map[n | n.toLowerCase].contains(declaration.name.toLowerCase)) {
+//            error("Duplicate declaration name: '" + declaration.name + "'",
+//                declaration.nameAttribute,
+//                DUPLICATE_DECLARATION_NAME,
+//                declaration.name)
+//        }
+//    }
```

Comment:
What is the reason of commented out check? Where is the alternative?

Reply:
The "def checkForDuplicateName(Named named)" changed to check any Named declaration.
The commented out code is temporary and will be removed later.